### PR TITLE
MIG-415, MIG 416: Direct image and PV migration

### PR DESCRIPTION
**Pull request by apinnick**
https://issues.redhat.com/browse/MIG-415
https://issues.redhat.com/browse/MIG-416

CP to 4.5, 4.6, 4.7.

The migration plan wizard has changed considerably, so you should check the preview build.

New "exposed route" field:
https://mig416-direct-image-migration--ocpdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/migrating-applications-with-cam-3-4.html#migration-adding-cluster-to-cam_migrating-3-4

Creating a migration plan:
https://mig416-direct-image-migration--ocpdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/migrating-applications-with-cam-3-4.html#migration-creating-migration-plan-cam_migrating-3-4

### CODE REVIEW

File: modules/migration-adding-cluster-to-cam.adoc [hunk 1]
```diff
@@ -40,6 +40,10 @@ eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiw
 * *Cluster name*: May contain lower-case letters (`a-z`) and numbers (`0-9`). Must not contain spaces or international characters.
 * *Url*: URL of the cluster's API server, for example, `\https://<master1.example.com>:8443`.
 * *Service account token*: String that you obtained from the source cluster.
+* *Exposed route to image registry*: Optional. You can specify a route to the image registry of your source cluster to enable direct migration for PVs and images, for example, `docker-registry-default.apps.cluster.com`.
```
Comment (XinRedhat) [inline modules/migration-adding-cluster-to-cam.adoc:43]: the Exposed route to image registry is to enable direct migration for images only.

File: modules/migration-creating-migration-plan-cam.adoc [hunk 2]
```diff
@@ -33,37 +33,48 @@ The *Plan name* can contain up to 253 lower-case alphanumeric characters (`a-z,
...
 
-* *Copy* copies the data in a source cluster's PV to the replication repository and then restores it on a newly created PV, with similar characteristics, in the target cluster.
+. Select a *Source cluster*, a *Target cluster*, and a *Repository*, and click *Next*.
+. In the *Namespaces* screen, select the projects to be migrated and click *Next*.
+. In the *Persistent volumes* screen, select *Copy* or *Move* for the PVs:
+
+* *Copy* copies the data from the PV of a source cluster to the replication repository and then restores it on a newly created PV, with similar characteristics, in the target cluster.
 +
-Optional: You can verify data copied with the file system method by selecting *Verify copy*. This option generates a checksum for each source file and checks it after restoration. The operation significantly reduces performance.
+If you specified a route to an image registry when you added the source cluster to the web console, you can migrate data directly from the source cluster to the target cluster.
```
Comment (XinRedhat) [inline modules/migration-creating-migration-plan-cam.adoc:42]: @apinnick I am a little confused about this words. I guess the meaning is 
"If you specified **a route of image registry** when you add the source cluster to the web console, you can migrate the **referenced images** directly from source cluster to the target cluster"

**Comment by XinRedhat** [comment]
@apinnick please see the comments

**Comment by apinnick** [comment]
@XinRedhat I made the changes you requested.

**Comment by XinRedhat** [comment]
@apinnick  LGTM

File: modules/migration-adding-cluster-to-cam.adoc [hunk 3]
```diff
@@ -40,6 +40,10 @@ eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiw
 * *Cluster name*: May contain lower-case letters (`a-z`) and numbers (`0-9`). Must not contain spaces or international characters.
 * *Url*: URL of the cluster's API server, for example, `\https://<master1.example.com>:8443`.
 * *Service account token*: String that you obtained from the source cluster.
+* *Exposed route to image registry*: Optional. You can specify a route to the image registry of your source cluster to enable direct migration for images, for example, `docker-registry-default.apps.cluster.com`.
```
Comment (bmcelvee) [inline modules/migration-adding-cluster-to-cam.adoc:43]: Suggestion: s/Exposed/Expose

File: modules/migration-creating-migration-plan-cam.adoc [hunk 4]
```diff
@@ -27,37 +27,48 @@ The *Plan name* can contain up to 253 lower-case alphanumeric characters (`a-z,
 . Select the projects to be migrated and click *Next*.
 . Select *Copy* or *Move* for the persistent volume (PV):
 
-* *Copy* copies the data in a source cluster's PV to the replication repository and then restores it on a newly created PV, with similar characteristics, in the target cluster.
+. Select a *Source cluster*, a *Target cluster*, and a *Repository*, and click *Next*.
+. In the *Namespaces* screen, select the projects to be migrated and click *Next*.
```
Comment (bmcelvee) [inline modules/migration-creating-migration-plan-cam.adoc:31]: Suggestion: add colons after *Copy* and *Move*

**Comment by bmcelvee** [comment]
Just a couple of suggestions, otherwise LGTM!

File: modules/migration-adding-cluster-to-cam.adoc (hunk 3 above)
Comment (apinnick) [reply modules/migration-adding-cluster-to-cam.adoc:43]: No. That changes the meaning. 

First, the markup indicates that this is the UI. Documentation must follow the UI, not vice versa. 

Second, the field refers to an "exposed route". It's not a procedure step. The user has exposed a route to the image registry and needs to enter the route in this field.

File: modules/migration-creating-migration-plan-cam.adoc (hunk 4 above)
Comment (apinnick) [reply modules/migration-creating-migration-plan-cam.adoc:31]: No. *Copy* and *Move* are the subjects of the verb "copies" and make grammatical sense. There is no reason to turn these sentences into fragments.

**Comment by apinnick** [comment]
@bmcelvee  Comments addressed. Please merge and CP.

**Pull request merged**